c7n/resources/asg.py

```diff
@@ -492,6 +493,52 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('launch-activity')
+class LaunchActivityFilter(Filter):
+    """Filter ASG based on launch activity status
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: asg-launch-failures-weekly
+                resource: asg
+                filters:
+                  - launch-activity
+                    status: Failed
+                    days: 7
+    """
+    schema = type_schema(
+        'launch-activity',
+        status={'type': 'string'},
+        days={'type': 'number'})
+
+    @worker
+    def asg_activities(self, asg):
+        if 'c7n.Activities' in asg:
```

Comment:
put a different way what if i want to AND or OR on asgs with two asg events, i'd have two different filters using this, and it would always fail a second filter, because it only annotates with events that match the first filter.